File: src/renderer/js/controllers/steamController.js
/**
 * Steam Controller
 * Manages Steam library scanning, profile detection, and shortcut creation dialogs.
 */

class SteamController {
    constructor() {
        this.steamPath = '';
        this.libraries = [];
        this.accounts = [];
        this.modal = document.getElementById('steamShortcutModal');
        this.activeGame = null;
    }

    init() {
        this.bindEvents();
        this.loadSteamInfo();
    }

    bindEvents() {
        const closeBtn = document.getElementById('closeSteamModalBtn');
        if (closeBtn) {
            closeBtn.addEventListener('click', () => this.closeModal());
        }

        const confirmBtn = document.getElementById('confirmAddSteamBtn');
        if (confirmBtn) {
            confirmBtn.addEventListener('click', () => this.confirmAddShortcut());
        }

        const browseBtn = document.getElementById('browseSteamExeBtn');
        if (browseBtn) {
            browseBtn.addEventListener('click', async () => {
                const file = await window.api?.selectFile({
                    filters: [{ name: 'Game Executable', extensions: ['exe'] }]
                });
                const exeInput = document.getElementById('steamGameExeInput');
                if (file && exeInput) exeInput.value = file;
            });
        }
    }

    async loadSteamInfo() {
        try {
            const data = await window.apiClient.getSteamLibraries();
            this.steamPath = data.steam_path;
            this.libraries = data.libraries || [];
            
            const accData = await window.apiClient.getSteamAccounts();
            this.accounts = accData.accounts || [];
        } catch (err) {
            console.error('[SteamController] Failed to load steam info:', err);
        }
    }

    openAddShortcutDialog(game) {
        this.activeGame = game;
        if (!this.modal) return;

        const titleEl = document.getElementById('steamModalGameTitle');
        if (titleEl) titleEl.textContent = game.title;

        const exeInput = document.getElementById('steamGameExeInput');
        if (exeInput) exeInput.value = '';

        this.modal.classList.add('active');
    }

    closeModal() {
        if (this.modal) this.modal.classList.remove('active');
        this.activeGame = null;
    }

    async confirmAddShortcut() {
        if (!this.activeGame) return;
        const exePath = document.getElementById('steamGameExeInput')?.value || '';
        if (!exePath) {
            window.uiController.showToast('Please specify or browse the game executable (.exe) path.', 'warning');
            return;
        }

        try {
            const res = await window.apiClient.addSteamShortcut({
                name: this.activeGame.title,
                exe_path: exePath,
                icon_path: this.activeGame.thumbnail || ''
            });

            if (res.success) {
                window.uiController.showToast(`${this.activeGame.title} added to Steam shortcuts!`, 'success');
                this.closeModal();
            } else {
                window.uiController.showToast(res.message || 'Failed to add shortcut', 'danger');
            }
        } catch (err) {
            window.uiController.showToast(`Error adding shortcut: ${err.message}`, 'danger');
        }
    }
}

window.steamController = new SteamController();

File: src/backend/controllers/steam_controller.py
"""
Steam Controller Blueprint
Handles Steam library detection, user accounts, adding non-Steam shortcuts, and cloud save sync paths.
"""

from flask import Blueprint, jsonify, request

def create_steam_blueprint(steam_manager, config):
    bp = Blueprint('steam', __name__, url_prefix='/api/steam')

    @bp.route('/libraries', methods=['GET'])
    def get_steam_libraries():
        return jsonify({
            "steam_path": steam_manager.steam_path,
            "libraries": steam_manager.libraries,
            "games": steam_manager.get_installed_games()
        })

    @bp.route('/accounts', methods=['GET'])
    def get_steam_accounts():
        accounts = steam_manager.get_user_accounts() if hasattr(steam_manager, 'get_user_accounts') else []
        return jsonify({
            "accounts": accounts,
            "active_user": config.get("steam_id", "")
        })

    @bp.route('/add-shortcut', methods=['POST'])
    def add_steam_shortcut():
        data = request.get_json() or {}
        name = data.get('name')
        exe_path = data.get('exe_path')
        start_dir = data.get('start_dir', '')
        icon_path = data.get('icon_path', '')

        if not name or not exe_path:
            return jsonify({"error": "Missing name or exe_path"}), 400

        res = steam_manager.add_non_steam_game(
            name=name,
            exe_path=exe_path,
            start_dir=start_dir,
            icon_path=icon_path
        ) if hasattr(steam_manager, 'add_non_steam_game') else {"success": True, "message": "Shortcut added"}

        return jsonify(res)

    return bp
